### PR TITLE
Support tiflash

### PR DESCRIPTION
**Pull request by yeya24**
Add tiflash support. Since this PR is kind of large, I will open another PR for chaos for tiflash

### CODE REVIEW

File: pkg/test-infra/tiflash/operation.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ngaut/log"
+	"github.com/pingcap/errors"
+	clusterTypes "github.com/pingcap/tipocket/pkg/cluster/types"
+	"github.com/pingcap/tipocket/pkg/test-infra/fixture"
+	"github.com/pingcap/tipocket/pkg/test-infra/tidb"
+	"github.com/pingcap/tipocket/pkg/test-infra/util"
```
Comment (mahjonp) [inline pkg/test-infra/tiflash/operation.go]: please group imports here

File: pkg/test-infra/tiflash/operation.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+// Apply TiFlash cluster
+func (o *Ops) Apply(tc *Recommendation) error {
+	// config file initialization
+	if fixture.Context.TiKVConfigFile == "" {
+		fixture.Context.TiKVConfigFile = "/config/tiflash/tikv.toml"
+	}
+	if fixture.Context.PDConfigFile == "" {
+		fixture.Context.PDConfigFile = "/config/tiflash/pd.toml"
```
Comment (mahjonp) [inline pkg/test-infra/tiflash/operation.go]: Setting fixture.Context here isn't recommended, and "/config/tiflash/pd.toml" will confuse someone unfamiliar with Tipocket.

Comment (mahjonp) [reply pkg/test-infra/tiflash/operation.go]: How about rename the configmap dir to config on this PR?

File: pkg/test-infra/tiflash/operation.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+	return append(tidbClusterNodes, tiFlashNode), nil
+}
+
+// GetTiFlashNode returns the nodes of TiFlash
+func (o *Ops) GetTiFlashNode(TiFlash *TiFlash) (clusterTypes.Node, error) {
+	pod := &corev1.Pod{}
+	objKey := client.ObjectKey{
+		Namespace: TiFlash.StatefulSet.ObjectMeta.Namespace,
+		Name:      fmt.Sprintf("%s-0", TiFlash.StatefulSet.ObjectMeta.Name),
```
Comment (mahjonp) [inline pkg/test-infra/tiflash/operation.go]: maybe we need retrieve all tiflash pods?

File: pkg/test-infra/tiflash/template.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    [security]
+    ca-path = ""
+    cert-path = ""
+    key-path = ""
+
+    [import]
+
+`))
+
+func renderTemplateFunc(tpl *template.Template, model interface{}) (string, error) {
```
Comment (mahjonp) [inline pkg/test-infra/tiflash/template.go]: How about extract it into a public method? I see many of renderTemplateFunc on our codebase.

**Comment by yeya24** [comment]
@mahjonp  I have addressed all of your comments. PTAL.

File: config/tiflash/pd.toml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[log]
+level = "info"
+[replication]
+location-labels = ["region", "zone", "rack", "host"]
+enable-placement-rules = true
+[schedule]
+merge-schedule-limit = 0
```
Comment (lidezhu) [inline config/tiflash/pd.toml]: No need to set merge-schedule-limt to 0 anymore.

File: config/tiflash/tikv.toml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+log-level = "info"
+[raftstore]
+hibernate-regions = false
```
Comment (lidezhu) [inline config/tiflash/tikv.toml]: No need to set hibernate-regions to false anymore.

**Comment by yeya24** [comment]
Thanks @lidezhu I have updated my code. Please give a another review in your free time.

**Pull request merged**